finality-aleph/src/session_map.rs

```diff
@@ -74,19 +79,32 @@ where
 impl<C, B, BE> AuthorityProvider for AuthorityProviderImpl<C, B, BE>
 where
     C: ClientForAleph<B, BE> + Send + Sync + 'static,
-    C::Api: crate::aleph_primitives::AlephSessionApi<B>,
+    C::Api: AlephSessionApi<B> + AuraApi<B, AuraId>,
     B: Block<Hash = BlockHash>,
     B::Header: Header<Number = BlockNumber>,
     BE: Backend<B> + 'static,
 {
+    fn aura_authorities(&self, block_number: BlockNumber) -> Option<Vec<AuraId>> {
+        AuraApi::authorities(
+            self.client.runtime_api().deref(),
+            self.block_hash(block_number)?,
+        )
+        .ok()
+    }
+
+    fn next_aura_authorities(&self, block_number: BlockNumber) -> Option<Vec<AuraId>> {
+        AlephSessionApi::next_session_aura_authorities(
+            self.client.runtime_api().deref(),
+            self.block_hash(block_number)?,
+        )
+        .ok()
+    }
+
     fn authority_data(&self, block_number: BlockNumber) -> Option<SessionAuthorityData> {
         let block_hash = self.block_hash(block_number)?;
         match self.client.runtime_api().authority_data(block_hash) {
             Ok(data) => Some(data),
-            Err(_) => self
```

Comment:
hmm, why is this better?

Reply:
it's necessary cause both `AlephSessionApi` and `AuraApi` have a method called `authorities`, right?

Reply:
you re right